src/streams/appendToStream/batchAppend.ts

```diff
@@ -40,72 +45,76 @@ export const batchAppend = async function (
     StreamsClient,
     "appendToStream",
     (client) =>
-      client.batchAppend(
-        ...this.callArguments(baseOptions, {
-          deadline: Infinity,
-        })
-      ),
-    streamCache
-  )();
-
-  return new Promise((resolve, reject) => {
-    const detach = () => {
-      stream.off("data", handleData).off("error", handleError);
-    };
-
-    const handleData = (resp: BatchAppendResp) => {
-      const resultingId = parseUUID(resp.getCorrelationId());
-      if (correlationId !== resultingId) return;
-
-      detach();
-
-      if (resp.hasError()) {
-        const grpcError = resp.getError()!;
-
-        if (!this.throwOnAppendFailure) {
-          const wrongExpectedVersion = unpackWrongExpectedVersion(grpcError);
-
-          if (wrongExpectedVersion) {
-            const nextExpectedRevision =
-              wrongExpectedVersion.hasCurrentStreamRevision()
-                ? BigInt(wrongExpectedVersion.hasCurrentStreamRevision())
-                : BigInt(-1);
-
-            return resolve({
-              success: false,
-              nextExpectedRevision,
-              position: undefined,
-            });
+      client
+        .batchAppend(
+          ...this.callArguments(baseOptions, {
+            deadline: Infinity,
+          })
+        )
+        .on("data", (resp: BatchAppendResp) => {
+          const resultingId = parseUUID(resp.getCorrelationId()!);
```

Comment:
**QUESTION:** Is it possible that correlation id won't be available?

Reply:
No, but everything is optional in the generated grpc code, so we need to use `!` here.